ogr/ogrgeomfielddefn.cpp

```diff
@@ -123,6 +123,49 @@ OGRGeomFieldDefn::~OGRGeomFieldDefn()
         const_cast<OGRSpatialReference *>(poSRS)->Release();
 }
 
+/************************************************************************/
+/*                          OGRGeomFieldDefn::OGRGeomFieldDefn()        */
+/************************************************************************/
+
+/**
+ * @brief OGRGeomFieldDefn::OGRGeomFieldDefn Copy constructor
+ * @param oOther the OGRGeomFieldDefn to copy.
+ * @since GDAL 3.11
+ */
+OGRGeomFieldDefn::OGRGeomFieldDefn(const OGRGeomFieldDefn &oOther)
+    : pszName(CPLStrdup(oOther.pszName)), eGeomType(oOther.eGeomType),
+      poSRS(nullptr), bNullable(oOther.bNullable), m_bSealed(false),
+      m_oCoordPrecision(oOther.m_oCoordPrecision)
+{
+    if (oOther.poSRS)
+    {
+        poSRS = oOther.poSRS->Clone();
+    }
+}
+
+/************************************************************************/
+/*                          OGRGeomFieldDefn::operator=()               */
+/************************************************************************/
+
+/**
+ * Copy assignment operator
+ * @param oOther the OGRGeomFieldDefn to copy.
+ * @return a reference to the current object.
+ * @since GDAL 3.11
+ */
+OGRGeomFieldDefn &OGRGeomFieldDefn::operator=(const OGRGeomFieldDefn &oOther)
+{
+    if (&oOther != this)
+    {
+        SetName(oOther.pszName);
+        SetType(oOther.eGeomType);
+        SetSpatialRef(oOther.poSRS);
```

Comment:
Looking at SetSpatialRef() implementation I can see it could have an issue if this.poSRS == oOther.poSRS and the reference count is 1

The method should be modified to:
```
void OGRGeomFieldDefn::SetSpatialRef(const OGRSpatialReference *poSRSIn)
{
    if (m_poSRS == poSRSIn)
    {
        return;
    }
    if (m_bSealed)
    {
        CPLError(
            CE_Failure, CPLE_AppDefined,
            "OGRGeomFieldDefn::SetSpatialRef() not allowed on a sealed object");
        return;
    }
    if (poSRS != nullptr)
        const_cast<OGRSpatialReference *>(poSRS)->Release();
    poSRS = poSRSIn;
    if (poSRS != nullptr)
        const_cast<OGRSpatialReference *>(poSRS)->Reference();
}
```

Reply:
done